### PR TITLE
Dashboard filters

### PR DESCRIPTION
**Pull request by Schrammel**
## Description

Improve performance (FPS) on dashboard transitions and implements a simple filter/sorter

### CODE REVIEW

File: apps/cave/components/Marketplace/Main/UseMarkeplaceState.ts [hunk 1]
```diff
@@ -41,12 +40,7 @@ export const useMarketplaceDashbord = () => {
   const [sort, setSort] = useState<NftSort>({ sort: 'REDEEM_DATE', order: 'ASC' })
   const sortFunction = sort ? positionSorter.data?.[sort.sort][sort.order] : () => 0
 
-  const [stakeFilters, setStakeFilters] = useState([
-    StakePoolFilterEnum['45_DAYS'],
-    StakePoolFilterEnum['180_DAYS'],
-    StakePoolFilterEnum['180_DAYS'],
-    StakePoolFilterEnum['360_DAYS'],
-  ])
+  const [stakeFilters, setStakeFilters] = useState<StakingPool[]>([...stakingPools])
```
Comment (greg-schrammel) [inline apps/cave/components/Marketplace/Main/UseMarkeplaceState.ts:43]: need to spread to a new array?  
```suggestion
  const [stakeFilters, setStakeFilters] = useState<StakingPool[]>(stakingPools)
```

File: apps/cave/components/NftFilters/Filters/StakePoolFilter.tsx [hunk 2]
```diff
@@ -1,86 +1,61 @@
...
+              const key = item.poolId.toString()
+              if (item.poolId === undefined) return <></>
+              return (
+                <MenuItemOption key={key} value={key} minH="48px">
+                  <HStack
+                    onClick={() => {
+                      const has = stakePoolFilters.some((p) => p.poolId === item.poolId)
+                      const removeList = [
+                        ...stakePoolFilters.filter((p: StakingPool) => p.poolId !== item.poolId),
+                      ]
```
Comment (greg-schrammel) [inline apps/cave/components/NftFilters/Filters/StakePoolFilter.tsx:42]: filter already returns a new array
```suggestion
  const removeList = stakePoolFilters.filter((p: StakingPool) => p.poolId !== item.poolId)
```

File: apps/cave/components/NftFilters/Filters/StakePoolFilter.tsx [hunk 3]
```diff
@@ -1,86 +1,61 @@
...
+                      setStakePoolFilters(has ? removeList : [...stakePoolFilters, item])
```
Comment (greg-schrammel) [inline apps/cave/components/NftFilters/Filters/StakePoolFilter.tsx:43]: can we make `stakePoolFilters` a `Map`
```ts
return stakePoolFilters.has(poolId) 
? stakePoolFilter.remove(poolId) 
:  stakePoolFilter.add(item.poolId, item)  

```

File: apps/cave/components/NftFilters/Filters/StakePoolFilter.tsx [hunk 4]
```diff
@@ -1,86 +1,61 @@
...
-            apply={'background.glass'}
-            css={{ '::after': { opacity: 0.9 } }}
-            overflow="hidden"
-            transition={'0.3s all'}
-            pt={4}
-            gap={1}
+        <MenuList minWidth="240px">
+          <MenuOptionGroup
+            value={stakePoolFilters.map((i) => i.poolId.toString())}
+            defaultValue={[...stakePoolFilters.map((i) => i.toString())]}
```
Comment (greg-schrammel) [inline apps/cave/components/NftFilters/Filters/StakePoolFilter.tsx:29]:
```suggestion
            defaultValue={stakePoolFilters.map((i) => i.toString())}
```

File: libs/marketplace/src/fetcher.ts [hunk 5]
```diff
@@ -20,8 +20,10 @@ export const listUserHistory = ({
 export const listPositons = async ({
```
Comment (greg-schrammel) [inline libs/marketplace/src/fetcher.ts:20]:
```suggestion
export const listPositions = async ({
```

File: libs/marketplace/src/fetcher.ts [hunk 6]
```diff
@@ -31,7 +33,7 @@ export const listPositons = async ({
     .filter((l) => !owner || l.to.toLocaleLowerCase() === owner.toLocaleLowerCase())
   const { stakingV1ToStakingPosition } = parser(stakingV1Contract, provider)
   const result = await Promise.all(preFilter.map(stakingV1ToStakingPosition))
-  return result.filter((p) => !p.currentValue.eq(0)) //remove redeemeds
+  return result.filter((p) => excludeRedeemed && !p.currentValue.eq(0)) //remove redeemeds
```
Comment (greg-schrammel) [inline libs/marketplace/src/fetcher.ts:36]: no need to filter if `excludeRedeemed ` is false
```suggestion
  return excludeRedeemed ? result.filter((p) => p.currentValue.eq(0)) : result
```

File: libs/icons/components/index.ts [hunk 7]
```diff
@@ -1,5 +1,5 @@
 export { AirdropIcon } from './AirdropIcon'
-export { DashboardIcon } from './DashboardIcon'
+export * from './ReactIcons'
```
Comment (greg-schrammel) [inline libs/icons/components/index.ts:2]: does this treeshake correctly or are all icons included in this page?

File: apps/cave/components/NftFilters/Filters/StakePoolFilter.tsx (hunk 3 above)
Comment (Schrammel) [reply apps/cave/components/NftFilters/Filters/StakePoolFilter.tsx]: i used a Set

**Review by greg-schrammel — APPROVED** [review]

**Pull request merged**